### PR TITLE
fix: Disabled state issue for components

### PR DESCRIPTION
**Pull request by albinAppsmith**
## Description

Fixed disabled state issues in button, dropdown and text input components.

Fixes [#17431](https://github.com/appsmithorg/appsmith/issues/17431)

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Tested in storybook

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/design-system/src/TextInput/TextInput.stories.tsx [hunk 1]
```diff
@@ -16,15 +16,13 @@ const Template: ComponentStory<typeof TextInput> = (args) => {
...
+  $padding: "8px",
   autoFocus: true,
   className: "t--commit-comment-input",
-  disabled: true,
-  fill: true,
-  height: "80px",
+  height: "36px",
   onChange: () => console.log("input has changed"),
-  placeholder: "This is a Text Input placeholder",
+  placeholder: "Enter text",
```
Comment (tanvibhakta) [inline packages/design-system/src/TextInput/TextInput.stories.tsx:24]: Good job on the relevant copy!

File: packages/design-system/src/TextInput/index.tsx [hunk 2]
```diff
@@ -222,6 +222,7 @@ export const InputWrapper = styled.div<{
   isValid?: boolean;
   disabled?: boolean;
   $isLoading?: boolean;
+  readOnly?: boolean;
```
Comment (tanvibhakta) [inline packages/design-system/src/TextInput/index.tsx:225]: If you're changing the API, where is the associated main repo PR? 

read-only text input is what UneditableTextField claims to be. Can you find out what the differences are between them and if they're the same replace all instances of UneditableTextField with them?

Comment (albinAppsmith) [reply packages/design-system/src/TextInput/index.tsx:225]: I'm not changing the API here, I'm just passing this variable to the wrapper in order to fix the style. The prop was already there for the component

Comment (tanvibhakta) [reply packages/design-system/src/TextInput/index.tsx:225]: Ah my bad. Approving.

**Review by tanvibhakta — APPROVED** [review]

**Comment by albinAppsmith** [comment]
@shadabbuchh fixed the dropdown issues

**Comment by albinAppsmith** [comment]
@shadabbuchh fixed the dropdown issues

**Comment by shadabbuchh** [comment]
@albinAppsmith the dropdown issue still exists.

**Comment by albinAppsmith** [comment]
@shadabbuchh  I have fixed this, can you re-validate this?

**Pull request merged**